opm/simulators/wells/WellGroupHelpers.hpp

```diff
@@ -296,33 +296,82 @@ namespace Opm {
+                if (currentGroupControl != Group::ProductionCMode::FLD) {
+                    continue;
+                }
+            }
+            for (int phase = 0; phase < np; phase++) {
+                pot[phase] += thisPot[phase];
+            }
         }
-        bool isInjector = group.isInjectionGroup();
-        bool isProducer = group.isProductionGroup();
+        for (const std::string& wellName : group.wells()) {
+            const auto& wellTmp = schedule.getWell(wellName, reportStepIdx);
 
-        if (!isInjector && !isProducer)
-            return;
+            if (wellTmp.isProducer() && isInjector)
+                continue;
+
+            if (wellTmp.isInjector() && !isInjector)
+                continue;
+
+            if (wellTmp.getStatus() == Well::Status::SHUT)
+                continue;
+            const auto& end = wellState.wellMap().end();
+            const auto& it = wellState.wellMap().find( wellName );
+            if (it == end)  // the well is not found
+                continue;
+
+            int well_index = it->second[0];
+            const auto wellrate_index = well_index * wellState.numPhases();
+            // add contribution from wells not under group control
+            if (isInjector) {
+                if (wellState.currentInjectionControls()[well_index] == Well::InjectorCMode::GRUP)
+                    for (int phase = 0; phase < np; phase++) {
+                        pot[phase] += wellState.wellPotentials()[wellrate_index + phase];
+                    }
+            } else {
+                if (wellState.currentProductionControls()[well_index] ==  Well::ProducerCMode::GRUP)
+                    for (int phase = 0; phase < np; phase++) {
+                        pot[phase] -= wellState.wellPotentials()[wellrate_index + phase];
+                    }
+            }
+        }
 
         double oilPot = 0.0;
         if (pu.phase_used[BlackoilPhases::Liquid])
-            oilPot = sumWellPhaseRates(wellState.wellPotentials(), group, schedule, wellState, reportStepIdx, pu.phase_pos[BlackoilPhases::Liquid], isInjector);
 
+            oilPot = pot [ pu.phase_pos[BlackoilPhases::Liquid]];
         double gasPot = 0.0;
         if (pu.phase_used[BlackoilPhases::Vapour])
-            gasPot = sumWellPhaseRates(wellState.wellPotentials(), group, schedule, wellState, reportStepIdx, pu.phase_pos[BlackoilPhases::Vapour], isInjector);
+            gasPot = pot [ pu.phase_pos[BlackoilPhases::Vapour]];
 
         double waterPot = 0.0;
         if (pu.phase_used[BlackoilPhases::Aqua])
-            waterPot = sumWellPhaseRates(wellState.wellPotentials(), group, schedule, wellState, reportStepIdx, pu.phase_pos[BlackoilPhases::Aqua], isInjector);
+            waterPot = pot [pu.phase_pos[BlackoilPhases::Aqua]];
 
-        guideRate->compute(group.name(), reportStepIdx, simTime, oilPot, gasPot, waterPot);
+        if (isInjector) {
```

Comment:
I admit I am confused by the difference here, I think I must look at the GuideRate class to understand properly.